onnxruntime/core/providers/cpu/tensor/where_op.cc

```diff
@@ -10,10 +10,29 @@
 
 namespace onnxruntime {
 // kernel builder functions
+#define WHERE_VERSIONED_TYPED_KERNEL_WITH_TYPE_NAME(type, type_name)               \
+  ONNX_CPU_OPERATOR_VERSIONED_TYPED_KERNEL(                                        \
+      Where,                                                                       \
+      9,                                                                           \
+      15,                                                                          \
+      type_name,                                                                   \
+      KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<type>()), \
+      Where<type>)
+
+#define WHERE_VERSIONED_TYPED_KERNEL(type) \
+  WHERE_VERSIONED_TYPED_KERNEL_WITH_TYPE_NAME(type, type)
+
+WHERE_VERSIONED_TYPED_KERNEL(uint8_t)
+WHERE_VERSIONED_TYPED_KERNEL(int32_t)
+WHERE_VERSIONED_TYPED_KERNEL(int64_t)
+WHERE_VERSIONED_TYPED_KERNEL(float)
+WHERE_VERSIONED_TYPED_KERNEL(double)
+WHERE_VERSIONED_TYPED_KERNEL_WITH_TYPE_NAME(std::string, string)
```

Comment:
if we are simply adding support for a BFloat16, do we need to update all the existing registrations for other types?

Reply:
There will be another PR by Hari. Change made here to make CI pass.

Reply:
I think the idea here is that we are not adding support for the BFloat16 type for the CPU kernel, but rather because Where got versioned on account of the BFloat16 support in ONNX, Liqun is versioning the (typed) kernels in ORT as well.